lib/codecs/src/encoding/format/otlp.rs

```diff
@@ -0,0 +1,131 @@
+use crate::encoding::ProtobufSerializer;
+use bytes::BytesMut;
+use opentelemetry_proto::proto::{
+    DESCRIPTOR_BYTES, LOGS_REQUEST_MESSAGE_TYPE, METRICS_REQUEST_MESSAGE_TYPE,
+    RESOURCE_LOGS_JSON_FIELD, RESOURCE_METRICS_JSON_FIELD, RESOURCE_SPANS_JSON_FIELD,
+    TRACES_REQUEST_MESSAGE_TYPE,
+};
+use tokio_util::codec::Encoder;
+use vector_config_macros::configurable_component;
+use vector_core::{config::DataType, event::Event, schema};
+use vrl::protobuf::encode::Options;
+
+/// Config used to build an `OtlpSerializer`.
+#[configurable_component]
+#[derive(Debug, Clone, Default)]
+pub struct OtlpSerializerConfig {
+    // No configuration options needed - OTLP serialization is opinionated
+}
+
+impl OtlpSerializerConfig {
+    /// Build the `OtlpSerializer` from this configuration.
+    pub fn build(&self) -> Result<OtlpSerializer, crate::encoding::BuildError> {
+        OtlpSerializer::new()
+    }
+
+    /// The data type of events that are accepted by `OtlpSerializer`.
+    pub fn input_type(&self) -> DataType {
+        DataType::Log | DataType::Metric | DataType::Trace
```

Comment:
```suggestion
        DataType::Log | DataType::Trace
```

If I'm reading this properly, OtlpSerializer::encode doesn't actually support metrics

Reply:
Yup, remnant from an old version. Deleted.